### PR TITLE
Feat/#340 - Taipy Core version control documentation

### PR DESCRIPTION
**Pull request by trgiangdo**
https://github.com/Avaiga/taipy-doc/issues/340

TODO:
- [x] Add a page for Legacy version migration
- [ ] Add example on how to use Taipy version control with other tools like Git, DVC, ...

### CODE REVIEW

File: docs/manuals/core/versioning/production_mode.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+In the example above, experiment "1.0" is push to production and now becomes a production version. The application is re-run and can access entities from older productions versions too.
+
+With out explicitly define the version, the latest version of your application is push to production.
+
+To remove a version from production version, you can run with `--delete-production-version` option with the version name on the command line interface.
+
+```console
+$ python arima_taipy_app.py --delete-production-version "1.0"
+```
+
```
Comment (trgiangdo) [inline docs/manuals/core/versioning/production_mode.md:91]: Add a section on create new production version with a new name

File: docs/manuals/core/versioning/production_mode.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
+In production environment, your Taipy Core application can access entities of multiple experiments.
+
+To push an experiment to production environment, you can run with `--production` option with the version name on the command line interface.
+
+```console
+$ python arima_taipy_app.py --production "1.0"
+```
+
+In the example above, experiment "1.0" is push to production and now becomes a production version. The application is re-run and can access entities from older productions versions too.
```
Comment (trgiangdo) [inline docs/manuals/core/versioning/production_mode.md]:
```suggestion
In the example above, experiment "1.0" is push to production and now becomes a production version. The application is re-run and can access entities from older production versions too.
```

File: docs/manuals/core/versioning/development_mode.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+When developing a Taipy Core application, data nodes, tasks, pipelines, ... or the execution flow are constantly changing.
```
Comment (joaoandre-avaiga) [inline docs/manuals/core/versioning/development_mode.md]: I don't get the `...`, is it missing some text?

Comment (trgiangdo) [reply docs/manuals/core/versioning/development_mode.md]:
```suggestion
When developing a Taipy Core application, data nodes, tasks, pipelines, etc. or the execution flow are constantly changing.
```
It's like 'etc'.

**Review by jrobinAV — COMMENTED** [review]
Globally, I like the content and the structure. Two remarks:
1. I believe the example and the way it is presented can be simplified. 
I am going to propose something directly on the PR.
2. I propose adding a section in `version-mgt.md` page dedicated to `help` option. I am going to propose something directly on the PR.

**Comment by jrobinAV** [comment]
Remaining work on my side:

- [x] review experiment mode page, 
- [x] review production mode page,
- [x] review migration page,
- [x] remove old .py files

**Comment by jrobinAV** [comment]
It is read for review. Please let me know your comments.

I would like to merge it soon probably tomorrow, so Vincent and the CS team can review it. You can still add comments, remarks, and suggestions even if the PR is already merged. I will apply your comments in another branch/PR if necessary.

File: docs/manuals/core/versioning/code_example/main.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import taipy.core as tp
+from taipy import Config
+
+
+def my_print_algo(entry: str):
+    # does nothing!
+    return entry
+
+
+input_cfg = Config.configure_data_node("input", default_data="a_sting")
```
Comment (trgiangdo) [inline docs/manuals/core/versioning/code_example/main.py]:
```suggestion
input_cfg = Config.configure_data_node("input", default_data="a_string")
```
A typo I suppose?

File: docs/manuals/core/versioning/code_example/main_with_changes.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import taipy.core as tp
+from taipy import Config
+
+
+def my_print_algo(entry: str):
+    # does nothing!
+    return entry
+
+
+input_cfg = Config.configure_data_node("input", default_data="a_sting")
```
Comment (trgiangdo) [inline docs/manuals/core/versioning/code_example/main_with_changes.py]:
```suggestion
input_cfg = Config.configure_data_node("input", default_data="a_string")
```

File: docs/manuals/core/versioning/experiment_mode.md [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+```console
+$ python main.py -l
+Version number                         Mode                   Creation date
+1.0                                    Experiment (latest)    2023-01-25 12:24:19
+325d0618-6f9e-459b-9597-48fa93a57a23   Experiment             2023-01-25 12:20:56
+26e56e84-9e7e-4a26-93f6-443d9aa541d9   Development            2023-01-25 12:20:33
+
+$ python main.py --experiment 1.0
+[2023-01-25 12:28:54,963][Taipy][INFO] job JOB_my_print_algo_2133dc18-643b-4351-872b-aedfc2c65c9c is completed.
+nb scenarios: 2
```
Comment (trgiangdo) [inline docs/manuals/core/versioning/experiment_mode.md:96]: When someone runs the code example that we provide, the number of scenarios is not shown on cli right? 

Can we have a better way to demonstrate this?

File: docs/manuals/core/versioning/experiment_mode.md [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+325d0618-6f9e-459b-9597-48fa93a57a23   Experiment             2023-01-25 12:20:56
+26e56e84-9e7e-4a26-93f6-443d9aa541d9   Development            2023-01-25 12:20:33
+```
+
+With `--experiment 1.0` option, an _experiment_ version is created and stored under
+the name "1.0".
+
+In this example, you can see the number of scenarios displayed is still 1. Taipy
+only considered the scenarios of the 1.0 version, and filters out the entities of
+the other versions.
```
Comment (trgiangdo) [inline docs/manuals/core/versioning/experiment_mode.md]:
```suggestion
other versions.
```

File: docs/manuals/core/versioning/experiment_mode.md [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+
+$ python main.py -l
+Version number                         Mode                   Creation date
+1.0                                    Experiment (latest)    2023-01-25 12:24:19
+325d0618-6f9e-459b-9597-48fa93a57a23   Experiment             2023-01-25 12:20:56
+26e56e84-9e7e-4a26-93f6-443d9aa541d9   Development            2023-01-25 12:20:33
+```
+
+As you can see, this time the number of scenarios displayed is 2. Indeed, we run
+twice the 1.0 version, so we have two scenarios attached to it. Scenarios from other
```
Comment (trgiangdo) [inline docs/manuals/core/versioning/experiment_mode.md]:
```suggestion
the 1.0 version twice, so we have two scenarios attached to it. Scenarios from other
```

File: docs/manuals/core/versioning/experiment_mode.md [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+versions are filtered out.
+
+# Clean entities of an existing version
+
+To run the _experiment_ version with a fresh start, you can run using _experiment_
```
Comment (trgiangdo) [inline docs/manuals/core/versioning/experiment_mode.md]:
```suggestion
To run an _experiment_ version with a fresh start, you can run on _experiment_
```

File: docs/manuals/core/versioning/experiment_mode.md [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+```console
+$ python main.py --experiment 1.0 --force
+[2023-01-25 12:55:05,484][Taipy][WARNING] The Configuration of version 1.0 is conflict with the current Python Config.
+[2023-01-25 12:52:05,692][Taipy][WARNING] Overriding version 1.0 ...
+[2023-01-25 12:52:05,890][Taipy][INFO] job JOB_my_print_algo_96ed74ed-183b-4dff-86c9-3b733d4d9bd9 is completed.
+nb scenarios: 2
+```
+
+As you can see on the previous example, the application run correctly after updating
+the configuration. A new scenario has been created submitted.
```
Comment (trgiangdo) [inline docs/manuals/core/versioning/experiment_mode.md]:
```suggestion
the configuration. A new scenario has been created.
```

**Review by trgiangdo — COMMENTED** [review]
Thank you @jrobinAV for the great updates. I think first version of the doc is ready for the CS team to review now.

There are a few comments, mostly typo and some are my questions. Then I think you can merge it and improve it later.

File: docs/manuals/core/versioning/experiment_mode.md (hunk 6 above)
Comment (jrobinAV) [reply docs/manuals/core/versioning/experiment_mode.md:96]: Yes it is. I added a print in the main.py file.

Comment (jrobinAV) [reply docs/manuals/core/versioning/experiment_mode.md:96]: here is a screenshot of my command line prompt:
image

Comment (trgiangdo) [reply docs/manuals/core/versioning/experiment_mode.md:96]: Forgive me if I miss it, but I can't find the print command in the code example that we provide

Comment (jrobinAV) [reply docs/manuals/core/versioning/experiment_mode.md:96]: OH right That s my bad. I forgot to update python files...

**Comment by jrobinAV** [comment]
I am merging it early so it can be reviewed by Vincent.

**Pull request merged**